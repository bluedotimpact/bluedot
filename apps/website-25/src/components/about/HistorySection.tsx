import { Section } from '@bluedot/ui';

const HistorySection = () => {
  return (
    <Section className="history-section" title="Our History">
      <div className="history-section__container flex flex-col gap-8 my-14">
        <HistoryEvent year="2021" event="A non-profit supporting students at the University of Cambridge to pursue high-impact careers" />
        <HistoryEvent year="2022" event="Growing into a new start-up called “BlueDot Impact” and moved headquarters to London" />
        <HistoryEvent year="2023" event="BlueDot Impact is now running 1 training course every 4 months and trains over xxx people in AI Safety and Biosecurity" />
        <HistoryEvent year="2024" event="BlueDot Impact is now running courses every 6 weeks and trains over 2500 professionals in AI Safety and Biosecurity" />
        <HistoryEvent year="2025" event="BlueDot is now running multiple courses in parallel every week! We are scaling the team to accelerate our impact!" now />
      </div>
    </Section>
  );
};

const HistoryEvent = ({ year, event, now }: { year: string, event: string, now?: boolean }) => {
  return (
    <div className={
        `history-section__event-container w-full flex flex-row gap-12 p-8 border-solid border-[1px] rounded-2xl
          ${now ? 'bg-bluedot-lighter border-bluedot-light' : 'border-gray-200'}`
      }
    >
      <strong className="history-section__year">{year}</strong>
      <p className="history-section__event-title">{event}</p>
    </div>
  );
};

export default HistorySection;
