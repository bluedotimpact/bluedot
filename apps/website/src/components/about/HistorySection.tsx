import { isMobile } from 'react-device-detect';
import { Section } from '@bluedot/ui';
import { P } from '../Text';

const HistorySection = () => {
  return (
    <Section className="history-section" title="Our history">
      <div className="history-section__container flex flex-col sm:flex-row sm:gap-2 gap-8">
        <HistoryEvent year="2021">Before BlueDot, we <strong>began as a University of Cambridge student group</strong>, hosting discussions on transformative emerging technologies.</HistoryEvent>
        <HistoryEvent year="2022"><strong>Launched our first AI Safety course</strong> with 400+ participants, <strong>founded BlueDot Impact</strong>, and established our <strong>London headquarters</strong>.</HistoryEvent>
        <HistoryEvent year="2023">Delivered a new course every four months, <strong>training 800+ people in AI Alignment, Governance, and Biosecurity</strong>.</HistoryEvent>
        <HistoryEvent year="2024">Accelerated to monthly courses, <strong>reaching 3,500+ participants worldwide</strong>. Over <strong>400</strong> of our course graduates now work in AI Safety.</HistoryEvent>
        <HistoryEvent year="2025" now>Launching multiple courses weekly and expanding a world-class team to <strong>scale our global impact</strong>.</HistoryEvent>
      </div>
    </Section>
  );
};

const HistoryEvent = ({ year, now, children }: { year: string, now?: boolean, children: React.ReactNode }) => {
  return (
    <div className="history-section__event flex-1">
      {isMobile ? (
        <div className={
          `history-section__event-container--mobile w-full flex flex-row gap-12 p-8
          ${now ? 'border bg-bluedot-lighter border-bluedot-light rounded-xl' : 'container-lined'}`
        }
        >
          <strong className="history-section__year">{year}</strong>
          <P className="history-section__event-details">{children}</P>
        </div>
      ) : (
        <div className="history-section__event-container--desktop flex flex-col gap-space-between">
          <div className="history-section__year-container w-full flex gap-2">
            <P className="history-section__year bg-bluedot-normal rounded-full px-4 py-2 text-color-text-on-dark text-[16px] font-bold w-min">{year}</P>
            <div className="history-section__year-decoration relative w-full after:content-[''] after:absolute after:top-1/2 after:w-full after:h-[2px] after:bg-bluedot-normal after:right-0" />
          </div>
          <P className="history-section__event-details mr-12">{children}</P>
        </div>
      )}
    </div>
  );
};

export default HistorySection;
