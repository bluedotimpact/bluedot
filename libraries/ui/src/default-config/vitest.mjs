/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
