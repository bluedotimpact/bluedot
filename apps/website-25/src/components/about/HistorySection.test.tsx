import { render } from '@testing-library/react';
import { describe, expect, test, vi } from 'vitest';
import * as deviceDetect from 'react-device-detect';
import HistorySection from './HistorySection';

describe('HistorySection', () => {
  test('renders desktop as expected', () => {
    const { container } = render(<HistorySection />);
    expect(container).toMatchSnapshot();
    expect(container.querySelector('.history-section__event-container--desktop')).not.toBeNull();
  });

  test('renders mobile as expected', () => {
    vi.spyOn(deviceDetect, 'isMobile', 'get').mockReturnValue(true);
    const { container } = render(<HistorySection />);
    expect(container).toMatchSnapshot();
    expect(container.querySelector('.history-section__event-container--mobile')).not.toBeNull();
    vi.clearAllMocks();
  });
});
