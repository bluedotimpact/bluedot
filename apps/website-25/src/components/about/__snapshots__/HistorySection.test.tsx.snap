// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HistorySection > renders desktop as expected 1`] = `
<div>
  <section
    class="section section-body history-section"
  >
    <div
      class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
    >
      <div
        class="section-heading__content flex-1 flex flex-col gap-2"
      >
        <h2
          class="section-heading__title relative"
        >
          Our history
        </h2>
      </div>
    </div>
    <div
      class="section__body"
    >
      <div
        class="history-section__container flex flex-col sm:flex-row sm:gap-2 gap-8"
      >
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--desktop flex flex-col gap-space-between"
          >
            <div
              class="history-section__year-container w-full flex gap-2"
            >
              <p
                class="history-section__year bg-bluedot-normal rounded-full px-4 py-2 text-on-dark text-[16px] font-bold w-min"
              >
                2021
              </p>
              <div
                class="history-section__year-decoration relative w-full after:content-[''] after:absolute after:top-1/2 after:w-full after:h-[2px] after:bg-bluedot-normal after:right-0"
              />
            </div>
            <p
              class="history-section__event-details mr-12"
            >
              Before BlueDot, we 
              <strong>
                began as a University of Cambridge student group
              </strong>
              , hosting discussions on transformative emerging technologies.
            </p>
          </div>
        </div>
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--desktop flex flex-col gap-space-between"
          >
            <div
              class="history-section__year-container w-full flex gap-2"
            >
              <p
                class="history-section__year bg-bluedot-normal rounded-full px-4 py-2 text-on-dark text-[16px] font-bold w-min"
              >
                2022
              </p>
              <div
                class="history-section__year-decoration relative w-full after:content-[''] after:absolute after:top-1/2 after:w-full after:h-[2px] after:bg-bluedot-normal after:right-0"
              />
            </div>
            <p
              class="history-section__event-details mr-12"
            >
              <strong>
                Launched our first AI Safety course
              </strong>
               with 400+ participants, 
              <strong>
                founded BlueDot Impact
              </strong>
              , and established our 
              <strong>
                London headquarters
              </strong>
              .
            </p>
          </div>
        </div>
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--desktop flex flex-col gap-space-between"
          >
            <div
              class="history-section__year-container w-full flex gap-2"
            >
              <p
                class="history-section__year bg-bluedot-normal rounded-full px-4 py-2 text-on-dark text-[16px] font-bold w-min"
              >
                2023
              </p>
              <div
                class="history-section__year-decoration relative w-full after:content-[''] after:absolute after:top-1/2 after:w-full after:h-[2px] after:bg-bluedot-normal after:right-0"
              />
            </div>
            <p
              class="history-section__event-details mr-12"
            >
              Delivered a new course every four months, 
              <strong>
                training 800+ people in AI Alignment, Governance, and Biosecurity
              </strong>
              .
            </p>
          </div>
        </div>
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--desktop flex flex-col gap-space-between"
          >
            <div
              class="history-section__year-container w-full flex gap-2"
            >
              <p
                class="history-section__year bg-bluedot-normal rounded-full px-4 py-2 text-on-dark text-[16px] font-bold w-min"
              >
                2024
              </p>
              <div
                class="history-section__year-decoration relative w-full after:content-[''] after:absolute after:top-1/2 after:w-full after:h-[2px] after:bg-bluedot-normal after:right-0"
              />
            </div>
            <p
              class="history-section__event-details mr-12"
            >
              Accelerated to monthly courses, 
              <strong>
                reaching 3,500+ participants worldwide
              </strong>
              . Over 
              <strong>
                400
              </strong>
               of our course graduates now work in AI Safety.
            </p>
          </div>
        </div>
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--desktop flex flex-col gap-space-between"
          >
            <div
              class="history-section__year-container w-full flex gap-2"
            >
              <p
                class="history-section__year bg-bluedot-normal rounded-full px-4 py-2 text-on-dark text-[16px] font-bold w-min"
              >
                2025
              </p>
              <div
                class="history-section__year-decoration relative w-full after:content-[''] after:absolute after:top-1/2 after:w-full after:h-[2px] after:bg-bluedot-normal after:right-0"
              />
            </div>
            <p
              class="history-section__event-details mr-12"
            >
              Launching multiple courses weekly and expanding a world-class team to 
              <strong>
                scale our global impact
              </strong>
              .
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`HistorySection > renders mobile as expected 1`] = `
<div>
  <section
    class="section section-body history-section"
  >
    <div
      class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
    >
      <div
        class="section-heading__content flex-1 flex flex-col gap-2"
      >
        <h2
          class="section-heading__title relative"
        >
          Our history
        </h2>
      </div>
    </div>
    <div
      class="section__body"
    >
      <div
        class="history-section__container flex flex-col sm:flex-row sm:gap-2 gap-8"
      >
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--mobile w-full flex flex-row gap-12 p-8
          container-lined"
          >
            <strong
              class="history-section__year"
            >
              2021
            </strong>
            <p
              class="history-section__event-details"
            >
              Before BlueDot, we 
              <strong>
                began as a University of Cambridge student group
              </strong>
              , hosting discussions on transformative emerging technologies.
            </p>
          </div>
        </div>
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--mobile w-full flex flex-row gap-12 p-8
          container-lined"
          >
            <strong
              class="history-section__year"
            >
              2022
            </strong>
            <p
              class="history-section__event-details"
            >
              <strong>
                Launched our first AI Safety course
              </strong>
               with 400+ participants, 
              <strong>
                founded BlueDot Impact
              </strong>
              , and established our 
              <strong>
                London headquarters
              </strong>
              .
            </p>
          </div>
        </div>
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--mobile w-full flex flex-row gap-12 p-8
          container-lined"
          >
            <strong
              class="history-section__year"
            >
              2023
            </strong>
            <p
              class="history-section__event-details"
            >
              Delivered a new course every four months, 
              <strong>
                training 800+ people in AI Alignment, Governance, and Biosecurity
              </strong>
              .
            </p>
          </div>
        </div>
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--mobile w-full flex flex-row gap-12 p-8
          container-lined"
          >
            <strong
              class="history-section__year"
            >
              2024
            </strong>
            <p
              class="history-section__event-details"
            >
              Accelerated to monthly courses, 
              <strong>
                reaching 3,500+ participants worldwide
              </strong>
              . Over 
              <strong>
                400
              </strong>
               of our course graduates now work in AI Safety.
            </p>
          </div>
        </div>
        <div
          class="history-section__event"
        >
          <div
            class="history-section__event-container--mobile w-full flex flex-row gap-12 p-8
          border bg-bluedot-lighter border-bluedot-light rounded-xl"
          >
            <strong
              class="history-section__year"
            >
              2025
            </strong>
            <p
              class="history-section__event-details"
            >
              Launching multiple courses weekly and expanding a world-class team to 
              <strong>
                scale our global impact
              </strong>
              .
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
